apps/src/code-studio/components/progress/teacherPanel/types.js

```diff
@@ -19,13 +19,19 @@ const lessonExtra = PropTypes.shape({
   nextLevelPath: PropTypes.string
 });
 
+// This is the shape of the data returned by summarize_for_teacher_panel in
+// script_level.rb.  Note that this data is also directly passed to code that
+// expects an object with shape levelWithProgressType (defined in
+// progress/progressTypes.js) so it must be compatible with that shape.
```

Comment:
This coupling worries me a bit but I didn't want to tackle it as part of this work (which has already taken much too much time).